lib/Sema/TypeCheckStmt.cpp

```diff
@@ -644,10 +642,8 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
   }
   
   Stmt *visitGuardStmt(GuardStmt *GS) {
-    StmtCondition C = GS->getCond();
-    TypeChecker::typeCheckStmtCondition(C, DC, diag::guard_always_succeeds);
-    GS->setCond(C);
-    
+    TypeChecker::typeCheckConditionForStatement(GS, DC);
```

Comment:
Can we leave this out of TypeChecker.h and internalize this function? There’s a lot of weird data dependencies here, so it’s okay if not.

Reply:
Internalize it in `TypeCheckStmt.cpp` file? Now that it's used from `NamingPatternRequest::evaluate` which is defined in `TypeCheckDecl.cpp`. I'd like to leave it for now.